docs/spikes/prism-v3/research/incentives-sources/calc.py

```
@@ -0,0 +1,47 @@
+
+print("="*70); print("1. COMPUTE BUDGET  (4x5090, peak %.0f TFLOPS)"%(PEAK/1e12))
+for h in (3.0,4.0,5.0,6.0):
+    row=[f"{C(m,h):.3e}" for m in (0.20,0.25,0.30,0.35)]
+    print(f"  {h:>4.1f}h  MFU20/25/30/35: {row}")
+
+print("\n2. COMPUTE-OPTIMAL N*  (C=6ND, r=D/N, N*=sqrt(C/6r))")
+def Nstar(Cv,r): return math.sqrt(Cv/(6.0*r))
+for Cv,lab in ((C(.25,4.0),"C@25%,4h"),(C(.30,4.0),"C@30%,4h"),(C(.30,5.0),"C@30%,5h"),(C(.35,5.0),"C@35%,5h")):
+    print(f"  {lab:<12} C={Cv:.3e}  " + "  ".join(f"r={r}:{Nstar(Cv,r)/1e6:6.1f}M" for r in (40,30,20,10,5,2,1)))
+
+print("\n   inverse: what r makes N=X optimal at C@30%,5h = %.2e"%C(.30,5.0))
```

Ruff (0.16.1)

[error] `%`-format string has unsupported format character `,`

(F509)